### DISCUSSION TITLE
Everything works, tests pass, but neovim crashes

### DISCUSSION THREAD

**Discussion by mrdgo** [tree-sitter/tree-sitter#3878] [Q&A] [answered]
In [my fork of tree-sitter-nu](https://github.com/mrdgo/tree-sitter-nu/tree/raw-strings-pt2) on branch `raw-strings-pt2`, I added support for raw strings. They start with `r#'` and end with `'#` - the twist is, they can have arbitrarily many `#`s, the only constraint is that the number on opening and closing must match. So I implemented a `scanner.c`. `tree-sitter generate; tree-sitter test` both run fine. I can also use the new scanner to query the tokens and highlight them in neovim. However, I only works in neovim-nightly, as soon as I run the exact same config on version `0.9.5`, `0.10.1`, or `0.10.2`, neovim crashes instantly. It crashes so hard that it doesn't even produce logs about it. Consider the output of `tail crash.log`:
```
finished sourcing /home/mrdgo/.local/share/bob/v0.10.2/share/nvim/runtime/plugin/tohtml.lua
continuing in nvim_exec2() called at /opt/dots/roles/nvim/files/nvim/init.lua:0
line 0: sourcing "/home/mrdgo/.local/share/bob/v0.10.2/share/nvim/runtime/plugin/tutor.vim"
finished sourcing /home/mrdgo/.local/share/bob/v0.10.2/share/nvim/runtime/plugin/tutor.vim
continuing in nvim_exec2() called at /opt/dots/roles/nvim/files/nvim/init.lua:0
line 0: sourcing "/home/mrdgo/.local/share/bob/v0.10.2/share/nvim/runtime/plugin/zipPlugin.vim"
finished sourcing /home/mrdgo/.local/share/bob/v0.10.2/share/nvim/runtime/plugin/zipPlugin.vim
continuing in nvim_exec2() called at /opt/dots/roles/nvim/files/nvim/init.lua:0
line 0: sourcing "/home/mrdgo/.local/share/nvim/lazy/tabular/after/plugin/TabularMaps.vim"
line 5: sourcing "/home/mrdgo/.local/sha
```
As you can see, it crashed while writing something to the logfile. I don't understand, why that is and would really appreciate some help.

Thanks in advance,
mrdgo

**Comment by amaanq** [comment]
use lldb to figure out what's going on.

**Reply by mrdgo** [reply]
Thanks for the reply, I tried using it. All I found is:
```
Process 643636 stopped
* thread #1, name = 'nvim', stop reason = signal SIGSEGV: address not mapped to object (fault address: 0x0)
    frame #0: 0x0000000000000000
error: memory read failed for 0x0
```
Apparently a null pointer dereference. How would you go on to find the cause?

**Reply by clason** [reply]
You need to build and test in Debug mode.

**Reply by mrdgo** [reply]
Build neovim myself?

**Reply by clason** [reply]
And try to capture a backtrace, see https://neovim.io/doc/user/dev_tools.html#dev-tools-backtrace

**Reply by mrdgo** [reply]
Build neovim or the parser with debug symbols?

**Reply by clason** [reply]
Both.

**Reply by mrdgo** [reply]
Thanks, will fiddle!

**Reply by mrdgo** [reply, marked as answer]
The problem disappeared. So thanks anyways :)
